landoapi/transplants.py

```diff
@@ -301,6 +303,47 @@ def warning_wip_commit_message(*, revision, **kwargs):
         return "This revision is marked as a WIP. Please remove `WIP:` before landing."
 
 
+@RevisionWarningCheck(8, "Repository is under a soft code freeze.", True)
+def warning_code_freeze(*, repo, **kwargs):
+    repo_fields = repo.get("fields", {})
+    short_name = repo_fields.get("shortName")
```

Comment:
Slightly cleaner:
```suggestion
    try:
        short_name = repo_fields["fields"]["shortName"]
    except KeyError:
        return
```

Or:
```suggestion
    if "fields" in repo_fields and "shortName" in repo_fields["fields"]:
        short_name = repo_fields["fields"]["shortName"]
    else:
        return
```

Although both those keys should always exist here, so I'm tempted to forego this prudence. Lots of things will break if `shortName` is missing from the repo. You should also consider using `PhabricatorClient.expect` which takes care of this.